### PR TITLE
Add audio feedback for text selection

### PR DESCRIPTION
**Pull request by nekodex**
https://user-images.githubusercontent.com/272140/185581073-0bd07257-a0ea-4750-aad2-78dd399f0b24.mov

- [x] depends on ppy/osu-resources#208
- [x] depends on ppy/osu-framework#5366
- [x] depends on https://github.com/ppy/osu/pull/19828 (framework bump fixes)

### CODE REVIEW

File: osu.Game/Graphics/UserInterface/OsuTextBox.cs [hunk 1]
```diff
@@ -112,7 +125,72 @@ protected override void OnCaretMoved(bool selecting)
...
+                    channel = selectWordSample?.GetChannel();
+                    break;
+
+                case TextSelectionType.Deselect:
+                    channel = deselectSample?.GetChannel();
+                    break;
+
+                default:
+                    channel = selectCharSample?.GetChannel();
+                    pitch += (SelectedText.Length / (float)Text.Length) * 0.15f;
```
Comment (Susko3) [inline osu.Game/Graphics/UserInterface/OsuTextBox.cs:184]: Possible division by 0 here, resulting in `NaN` pitch. Shockingly, this doesn't crash, but instead plays the sound high-pitched. It's probably not intended for selection to change and `Text.Length` to be `0` (framework-side bug) but I think this should be guarded against.

Also, pitch is `double`, but this is casting to `float`.

**Review by Susko3 — CHANGES_REQUESTED** [review]
Is it intended that both the `caretMoved` and `deselect` samples play when the selection is dismissed by keyboard arrow keys.

Eg. select some word in a middle of a sentence and then press left or right arrow key.

File: osu.Game/Graphics/UserInterface/OsuTextBox.cs (hunk 1 above)
Comment (nekodex) [reply osu.Game/Graphics/UserInterface/OsuTextBox.cs]: How were you able to trigger a selection change with `Text.Length` being zero? It's probably better to ensure `OnTextSelectionChanged` is never called when `Text.Length` is zero than guarding here?

**Comment by peppy** [comment]
Test failures here will be fixed via https://github.com/ppy/osu/pull/19828, marking as prereq for now.

**Review by peppy — APPROVED** [review]

**Pull request merged**